File: matiasmasso/Mat.NET/DT2/DTOClientFacturable.vb
Public Class DTOClientFacturable
    Property Customer As DTOCustomer
    Property Invoices As List(Of DTOInvoice)
    Property AlbaransPerFacturar As List(Of DTODelivery)
    Property Facturable As Boolean


    Public Sub New(Optional ByVal oCustomer As DTOCustomer = Nothing)
        MyBase.New
        _Customer = oCustomer
        _Invoices = New List(Of DTOInvoice)
        _AlbaransPerFacturar = New List(Of DTODelivery)
    End Sub

    Public Function Total() As DTOAmt
        Dim DcEur As Decimal = _Invoices.SelectMany(Function(x) x.Deliveries).Sum(Function(y) y.Import.Eur)
        Dim retval = DTOAmt.Factory(DcEur)
        Return retval
    End Function
End Class

File: matiasmasso/Mat.NET/DTO/DTOClientFacturable.cs
using System.Collections.Generic;
using System.Linq;

namespace DTO
{
    public class DTOClientFacturable
    {
        public DTOCustomer Customer { get; set; }
        public List<DTOInvoice> Invoices { get; set; }
        public List<DTODelivery> AlbaransPerFacturar { get; set; }
        public bool Facturable { get; set; }


        public DTOClientFacturable(DTOCustomer oCustomer = null/* TODO Change to default(_) if this is not a reference type */) : base()
        {
            Customer = oCustomer;
            Invoices = new List<DTOInvoice>();
            AlbaransPerFacturar = new List<DTODelivery>();
        }

        public DTOAmt Total()
        {
            decimal DcEur = Invoices.SelectMany(x => x.Deliveries).Sum(y => y.Import.Eur);
            var retval = DTOAmt.Factory(DcEur);
            return retval;
        }
    }
}
